airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -129,30 +130,30 @@ def _create_session_with_assume_role(self, session_kwargs: Dict[str, Any]) -> bo
             )
         session = botocore.session.get_session()
         session._credentials = credentials
+
         if self.basic_session is None:
             raise RuntimeError("The basic session should be created here!")
+
         region_name = self.basic_session.region_name
         session.set_config_variable("region", region_name)
+
         return boto3.session.Session(botocore_session=session, **session_kwargs)
 
     def _refresh_credentials(self) -> Dict[str, Any]:
         self.log.info('Refreshing credentials')
         assume_role_method = self.extra_config.get('assume_role_method', 'assume_role')
         sts_session = self.basic_session
+
+        if sts_session is None:
```

Comment:
We can create a basic_session with no extra arguments.

How about changing it so that basic_session is a "create+fash on first use" property?

Reply:
Not sure I understand -  basic_session is a "**create+fash** on first use" property?